miden/tests/integration/operations/field_ops.rs

```diff
@@ -1,12 +1,255 @@
+    test.expect_stack(&[7, c]);
+}
+
+#[test]
+fn add_b() {
+    let build_asm_op = |param: u64| format!("add.{}", param);
+
+    // --- simple case ----------------------------------------------------------------------------
+    let test = build_op_test!(build_asm_op(2), &[1]);
+    test.expect_stack(&[3]);
+
+    let test = build_op_test!(build_asm_op(8), &[5]);
+    test.expect_stack(&[13]);
+
+    // --- test overflow --------------------------------------------------------------------------
+    let test = build_op_test!(build_asm_op(8), &[Felt::MODULUS]);
+    test.expect_stack(&[8]);
+
+    // --- test that the rest of the stack isn't affected -----------------------------------------
+    let c = rand_value::<u64>();
+    let test = build_op_test!(build_asm_op(2), &[c, 5]);
+    test.expect_stack(&[7, c]);
+}
+
+#[test]
+fn sub() {
+    let asm_op = "sub";
+
+    // --- simple case ----------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[3, 2]);
+    test.expect_stack(&[1]);
+
+    let test = build_op_test!(asm_op, &[10, 7]);
+    test.expect_stack(&[3]);
+
+    // --- test underflow -------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[0, 1]);
+    test.expect_stack(&[Felt::MODULUS - 1]);
+
+    // --- test that the rest of the stack isn't affected -----------------------------------------
+    let c = rand_value::<u64>();
+    let test = build_op_test!(asm_op, &[c, 2, 2]);
+    test.expect_stack(&[0, c]);
+}
+
+#[test]
+fn sub_b() {
+    let build_asm_op = |param: u64| format!("sub.{}", param);
+
+    // --- simple case ----------------------------------------------------------------------------
+    let test = build_op_test!(build_asm_op(2), &[3]);
+    test.expect_stack(&[1]);
+
+    let test = build_op_test!(build_asm_op(7), &[10]);
+    test.expect_stack(&[3]);
+
+    // --- test underflow -------------------------------------------------------------------------
+    let test = build_op_test!(build_asm_op(1), &[0]);
+    test.expect_stack(&[Felt::MODULUS - 1]);
+
+    // --- test that the rest of the stack isn't affected -----------------------------------------
+    let c = rand_value::<u64>();
+    let test = build_op_test!(build_asm_op(2), &[c, 2]);
+    test.expect_stack(&[0, c]);
+}
+
+#[test]
+fn mul() {
+    let asm_op = "mul";
+
+    // --- simple cases ---------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[1, 0]);
+    test.expect_stack(&[0]);
+
+    let test = build_op_test!(asm_op, &[1, 5]);
+    test.expect_stack(&[5]);
+
+    // --- test overflow --------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[Felt::MODULUS, 2]);
+    let expected = Felt::MODULUS as u128 * 2 as u128 % Felt::MODULUS as u128;
+    test.expect_stack(&[expected as u64]);
```

Comment:
`Felt::MODULUS` gets reduced to `0` - so, this is basically testing multiplication by `0`. Maybe use `Felt::MODULUS - 1` instead?

Reply:
Oh right good catch.  Fixed